src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesHistory = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(stateCopy, action.extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const keyToRemove of action.keysToRemove) {
+          delete stateCopy[keyToRemove];
+        }
+        break;
+      }
+
+      case 'clear': {
+        const keysToClear = Object.keys(stateCopy);
+
+        for (const keyToClear of keysToClear) {
+          delete stateCopy[keyToClear];
+        }
+        break;
```

Comment:
we clear all values from our object so can we just assign new empty object to avoid additional loop?